pkg/apis/core/helper/helpers.go

```diff
@@ -532,3 +533,17 @@ func ToPodResourcesSet(podSpec *core.PodSpec) sets.String {
 	}
 	return result
 }
+
+// GetDeletionCostFromPodAnnotations returns the integer value of pod-deletion-cost. Returns 0
+// if not set or the value is invalid.
+func GetDeletionCostFromPodAnnotations(annotations map[string]string) (int64, error) {
+	if value, exist := annotations[core.PodDeletionCost]; exist {
+		i, err := strconv.ParseInt(value, 10, 64)
```

Comment:
An explicit plus sign (e.g. "+10") and leading zeros (e.g. "0080") are two surprising values this accepts. Suggest requiring this to be either exactly "0" or start with a non-zero digit.

Also, integer values > 52 bits are problematic in some languages, which is why most API integer fields are limited to 32 bit integers (e.g. pod priority uses int32). Suggest limiting this to 32 bits as well.

Reply:
> An explicit plus sign (e.g. "+10") and leading zeros (e.g. "0080") are two surprising values this accepts. Suggest requiring this to be either exactly "0" or start with a non-zero digit.

Why surprising? those are valid int values. Note that negative values are accepted, so "-10" is a valid value.

> Also, integer values > 52 bits are problematic in some languages, which is why most API integer fields are limited to 32 bit integers (e.g. pod priority uses int32). Suggest limiting this to 32 bits as well.

done.

Reply:
> > An explicit plus sign (e.g. "+10") and leading zeros (e.g. "0080") are two surprising values this accepts. Suggest requiring this to be either exactly "0" or start with a non-zero digit.
> 
> Why surprising? those are valid int values. Note that negative values are accepted, so "-10" is a valid value.

It seems likely that things written in languages other than go might interpret this annotation, and those are variations that could choke non-go string-to-int parsers (unexpected leading +, treating leading 0's as octal, etc).

Reply:
done.